Create first Django Project

DrEdi opened the pull request:
- Create first application for blog
- Add some templates
- Add urls

DrEdi committed:
Create first Django Project

- Create first application for blog
- Add some templates
- Add urls

DrEdi commented:
Ohhh
I hoped that `SKIP=flake8,pep257 git commit` have to work

DrEdi committed:
Fix .pre-commit.yaml

DrEdi committed:
Fix .pre-commit-config.yaml

- Add new hook
- Exclude django-created files from flake8

DrEdi committed:
Fix .pre-commit-config.yaml file

Now it excludes dirs and files that created by Django

DrEdi commented:
Ok, I really don't have idea why it works on my local but fails here

DrEdi committed:
Fix .pre-commit-config.yaml

Realised how must really work this task

webknjaz commented on .pre-commit-config.yaml:
```diff
@@ -9,11 +9,8 @@
-        - blog/
-        - main/
-        - manage.py
+      exclude: '[main/, manage.py, blog/]'
```
AFAIR this should be a regexp string

webknjaz commented on .pre-commit-config.yaml:
```diff
@@ -9,11 +9,8 @@
-        - main/
-        - manage.py
+      exclude: '[main/, manage.py, blog/]'
+      # args: [--max-line-length=120]
```
Why did you comment this out?

webknjaz commented on README.md:
```diff
@@ -1 +1 @@
-# django-girls-blog-DrEdi
\ No newline at end of file
+# django-girls-blog-DrEdi
```
This change looks pointless.
(Unless forced by pre-commit hooks.)

webknjaz commented:
I would rebase it on top of master before merging instead of creating merge commits :)
Regarding `exclude`.. you have to use a kind of regular expression in this option's value

webknjaz reviewed (changes requested):
Could you please try using an array for `exclude`? If this won't work, use regexp.
Please also note that generally it is not required to wrap strings with quotes in YAML.

webknjaz commented on main/templates/main/index.html:
```diff
@@ -0,0 +1,15 @@
+</head>
+<body>
+	{% for post in posts %}
+    <div>
```
Indentation of for-loop start, div-block and for-loop end is broken here.

webknjaz commented on main/templates/main/index.html:
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html>
+<head>
+	<title>Main</title>
```
Normally you'd want to use spaces for indentation

webknjaz commented on main/templates/main/index.html:
```diff
@@ -0,0 +1,15 @@
+    <div>
+        <p>published: {{ post.published_date }}</p>
+        <h1><a href="">{{ post.title }}</a></h1>
+        <p>{{ post.text|linebreaksbr }}</p>
```
filter-pipes are normally wrapped with spaces

webknjaz commented on main/models.py:
```diff
@@ -0,0 +1,19 @@
+
+class Post(models.Model):
+
+	author = models.ForeignKey('auth.User')
```
Please use spaces for indentation

webknjaz commented on README.md:
```diff
@@ -1,2 +1,3 @@
+
```
This addition is pointless

webknjaz commented on .pre-commit-config.yaml:
```diff
@@ -9,6 +9,8 @@
     - id: end-of-file-fixer
     - id: flake8
+      exclude: '[main/, manage.py, blog/]'
+      # args: [--max-line-length=120]
```
Feel free to use increased value of max-line-length option

webknjaz commented on .pre-commit-config.yaml:
```diff
@@ -9,6 +9,8 @@
     - id: double-quote-string-fixer
     - id: end-of-file-fixer
     - id: flake8
+      exclude: '[main/, manage.py, blog/]'
```
You **must not** exclude the actual src. You may only ignore db migrations.

webknjaz commented on .pre-commit-config.yaml:
```diff
@@ -17,3 +19,4 @@
   sha: master
   hooks:
     - id: pep257
+      exclude: '[main/, manage.py, blog/]'
```
Add `remove-tabs` to hooks. And see my comment above regarding excludes.

DrEdi committed:
Fix some files

Add markdown to README.md
Correct spaces in main/models.py
Correct spaces in main/templates/main/index.html
Comment pep257 because it prodused too many errors

DrEdi committed:
Add css

Add css file
Fix files that were blocked in previous pull-request

webknjaz commented on main/templates/main/base.html:
```diff
@@ -0,0 +1,24 @@
+        <div class="col-md-8">
+            {% block content %}
+            {% endblock %}
+            </div>
```
The indentation is odd here, plz fix

webknjaz commented on main/templates/main/base.html:
```diff
@@ -0,0 +1,24 @@
+    <div class="page-header">
+        <h1><a href="/">Django Girls Blog</a></h1>
+    </div>
+<div class="content container">
```
Why did you reduce indentation of this block? Please be consistent

webknjaz commented on .pre-commit-config.yaml:
```diff
@@ -9,11 +9,19 @@
     - id: trailing-whitespace
 
-- repo: https://github.com/FalconSocial/pre-commit-mirrors-pep257
+# - repo: https://github.com/FalconSocial/pre-commit-mirrors-pep257
```
I want you to have this hook enabled

webknjaz commented on manage.py:
```diff
@@ -0,0 +1,22 @@
+        # issue is really that Django is missing to avoid masking other
+        # exceptions on Python 2.
+        try:
+            import django
```
You may add `  # noqa: F401` comment in the end of this line to make flake8 not raise this error here.

webknjaz commented on main/tests.py:
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
```
You may comment it out for now.

DrEdi commented:
Everything was Ok when I commited in my local machine.
Could we marge it when I fix all the files and only after start to deploy in new branch?

DrEdi committed:
Fix files

Return pep257 to pre-commit
Fix spaces in base.html
Fix error with import in manage.py

webknjaz commented:
Merge will be available once you fix all the dips and tests will pass.

DrEdi committed:
Add docstrings

DrEdi committed:
Fix .travis.yml

webknjaz commented on .pre-commit-config.yaml:
```diff
@@ -17,3 +19,10 @@
   sha: master
   hooks:
     - id: pep257
+      args: [--ignore=D203]
```
Why did you disable this rule? 
Here's what [the doc states about D203](http://pep257.readthedocs.io/en/latest/error_codes.html):
```
1 blank line required before class docstring
```

DrEdi replied:
But there's another one `D211 No blank lines allowed before class docstring` :)
I had a conflict

webknjaz replied:
Oh I see, in such cases, when it is not obvious why do you do something it is a good practice to write a comment and explain why did you do this. Otherwise it the future someone may try "fixing"/removing it just because they think it's not how things should be working.

webknjaz commented on blog/__init__.py:
```diff
@@ -0,0 +1 @@
+"""Do not write anything here."""
```
This is not a docstring as it doesn't describe the module's purpose. In fact, it's not even required anymore for Python 3 envs.

webknjaz commented on blog/settings.py:
```diff
@@ -0,0 +1,122 @@
+"""
+Django settings for blog project.
```
I would leave just this line. Information below isn't particularly helpful or useful.

webknjaz commented on blog/urls.py:
```diff
@@ -0,0 +1,22 @@
+"""blog URL Configuration.
+
+The `urlpatterns` list routes URLs to views. For more information please see:
```
Again, this is useful when you first start working with django, but then it's a pointless noise you don't want to keep in your project.

webknjaz commented on blog/wsgi.py:
```diff
@@ -0,0 +1,16 @@
+"""
+WSGI config for blog project.
+
+It exposes the WSGI callable as a module-level variable named ``application``.
```
The same comment as above applies here.

webknjaz commented on main/__init__.py:
```diff
@@ -0,0 +1 @@
+"""Do not write anything here."""
```
This is pointless.

webknjaz commented on main/admin.py:
```diff
@@ -0,0 +1,5 @@
+"""Admin panel code is here."""
```
Please do not write useless descriptions stating obvious things like "the second line in this module starts with 'f'". Try taking point of view of a stranger looking at the module his first time and trying to figure out what's going on here.
Write a high-level description of what is the purpose of this module.

webknjaz commented on .gitignore:
```diff
@@ -87,3 +87,6 @@ ENV/
 .ropeproject
+
+#PyCharm
+.idea/
```
Good catch, you also need to ignore SQLite database data file.

webknjaz commented on main/apps.py:
```diff
@@ -0,0 +1,8 @@
+"""Apps are here."""
```
What apps?

webknjaz commented on main/models.py:
```diff
@@ -0,0 +1,24 @@
+"""All models are here."""
```
What for?

webknjaz commented on main/models.py:
```diff
@@ -0,0 +1,24 @@
+
+
+class Post(models.Model):
+    """Model that represent post in our blog."""
```
represent`s`. Mind your grammar :)

webknjaz commented on main/models.py:
```diff
@@ -0,0 +1,24 @@
+        self.save()
+
+    def __str__(self):
+        """Return title of post."""
```
No. Make title represent instance when interpolating into str

webknjaz commented on main/templates/main/index.html:
```diff
@@ -0,0 +1,12 @@
+            <div class="date">
+                <p>published: {{ post.published_date }}</p>
+            </div>
+            <h1><a href="post/{{ post.pk }}/">{{ post.title }}</a></h1>
```
Shouldn't you use `{% url %}` tag here?

webknjaz commented on main/urls.py:
```diff
@@ -0,0 +1,8 @@
+"""Urls are here."""
```
Which URLs? Point the difference south the other similar module.

webknjaz commented on main/views.py:
```diff
@@ -0,0 +1,17 @@
+from django.shortcuts import render, get_object_or_404
+from .models import Post
+
+# Create your views here.
```
You don't need this comment, do you?

webknjaz commented on main/views.py:
```diff
@@ -0,0 +1,17 @@
+
+
+def post_detail(request, pk):
+    """Show info about one post user choosed."""
```
*Nit: you have chosen

webknjaz commented on manage.py:
```diff
@@ -0,0 +1,22 @@
+"""Main file for manage your project."""
```
One ring to rule them all :)

webknjaz reviewed (changes requested):
You have committed your local database, you have to remove it from the repo.

DrEdi committed:
Fix docstrings, delete db

Delete Db from previous commit
Add comment to pre-coomit why we ignore D203
